packages/fxa-settings/src/pages/InlineRecoverySetup/index.tsx

```diff
@@ -0,0 +1,233 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { RouteComponentProps } from '@reach/router';
+import { logPageViewEvent } from '../../lib/metrics';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useFtlMsgResolver } from '../../models';
+import InputText from '../../components/InputText';
+import { ReactComponent as CopyIcon } from './recovery_code_copy.svg';
+import { ReactComponent as DownloadIcon } from './recovery_code_download.svg';
+import { ReactComponent as PrintIcon } from './recovery_code_print.svg';
+import { ReactComponent as RecoveryCodesGraphic } from '../Signin/SigninRecoveryCode/graphic_recovery_codes.svg';
+
+export type InlineRecoverySetupProps = {
+  isIos?: boolean;
+  recoveryCodes: Array<string>;
+  serviceName?: string;
+  showConfirmation: boolean;
+};
+
+const InlineRecoverySetup = ({
+  isIos,
+  recoveryCodes,
+  serviceName,
+  showConfirmation,
+}: InlineRecoverySetupProps & RouteComponentProps) => {
+  logPageViewEvent('inline-recovery-setup', {
+    entrypoint_variation: 'react',
+  });
+
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedInputTextLabel = ftlMsgResolver.getMsg(
+    'inline-recovery-backup-authentication-code',
+    'Backup authentication code'
+  );
+  const localizedCopyButtonTitle = ftlMsgResolver.getMsg('copy-button', 'Copy');
+  const localizedDownloadButtonTitle = ftlMsgResolver.getMsg(
+    'inline-recovery-download-button',
+    'Download'
+  );
+  const localizedPrintButtonTitle = ftlMsgResolver.getMsg(
+    'inline-recovery-print-button',
+    'Print'
+  );
+  const localizedGraphicAriaLabel = ftlMsgResolver.getMsg(
+    'inline-recovery-setup-image-description',
+    'Document that contains hidden text.'
+  );
+
+  /* TODO: - Add in copy/download/print/continue/cancel actions for all buttons
+   *       - Add in metrics for all events
+   *       - Add tests for all metrics
+   */
+  const printCode = () => {};
+  const copyCode = () => {};
+  const downloadCode = () => {};
+  const cancelSetup = () => {};
+  const continueSetup = () => {};
+
+  return (
+    <>
+      {showConfirmation ? (
+        <>
+          <div className="mb-4">
+            {serviceName ? (
+              <FtlMsg id="inline-recovery-confirmation-header">
+                <h1 className="card-header">
+                  Confirm backup authentication code{' '}
+                  <span className="card-subheader">{`to continue to ${serviceName}`}</span>
+                </h1>
+              </FtlMsg>
+            ) : (
+              <FtlMsg id="inline-recovery-confirmation-header-default">
+                <h1 className="card-header">
```

Comment:
If we start a `serviceName` list, we can change the hard-coded `account settings` to point to whatever we call the "default". I'm guessing when we hook this up functionally, `serviceName` won't ever be undefined since it'll have that default.

Could at a minimum set the default to equal `account settings`, then have your `{serviceName ? (` check against `{serviceName !== 'account settings'` instead? I'm guessing we need this `inline-recovery-confirmation-header-default` separated because we want "account settings" localized vs not needing product names localized.

Reply:
Correct re: account settings! I can fix this up once we've merged in Signin, since that will have the new shared RP type

Reply:
I rebased this on top of the latest additions to main, and it uses the `serviceName` type now 👍